src/day7.scala

```diff
@@ -0,0 +1,76 @@
+// using resourceDir "../input"
+// using lib com.lihaoyi::pprint:0.6.6
+
+package day7
+
+import scala.util.Using
+import scala.io.Source
+import scala.annotation.tailrec
+
+extension (s: String)
+  def input: List[Int] =
+    Using.resource(Source.fromFile("input/day7"))(_.mkString).split(",").map(_.toInt).toList
+
+case class Crabmarine(
+    horizontal: Int,
+    increasingCost: Boolean,
+    fuelCost: Int = 1
+):
+  def moveUp() =
+    if increasingCost then
+      this.copy(horizontal = horizontal + 1, fuelCost = fuelCost + 1)
+    else this.copy(horizontal = horizontal + 1)
+  def moveDown() =
+    if increasingCost then
+      this.copy(horizontal = horizontal -1, fuelCost = fuelCost + 1)
+    else this.copy(horizontal = horizontal - 1)
+
+case class Crabmada(crabmarines: List[Crabmarine]):
+
+  require(crabmarines.nonEmpty)
+
+  @tailrec
+  final def align(
+      situation: List[Crabmarine] = crabmarines,
+      fuelCost: Int = 0
+  ): Int =
+    val allTheSame = situation.forall(_.horizontal == situation.head.horizontal)
+    if allTheSame then fuelCost
+    else
+      val topHorizontal = situation.maxBy(_.horizontal)
+      val minHorizontal = situation.minBy(_.horizontal)
+
+      val fuelCostForTop = situation.collect {
+        case p if p.horizontal == topHorizontal.horizontal => p.fuelCost
+      }.sum
+      val fuelCostForMin = situation.collect {
+        case p if p.horizontal == minHorizontal.horizontal => p.fuelCost
+      }.sum
+      if fuelCostForTop < fuelCostForMin then
+        val updated = situation.map { p =>
+          if p.horizontal == topHorizontal.horizontal then p.moveDown()
+          else p
+        }
+        align(updated, fuelCost + fuelCostForTop)
+      else
+
+        val updated = situation.map { p =>
+          if p.horizontal == minHorizontal.horizontal then p.moveUp()
+          else p
+        }
+        align(updated, fuelCost + fuelCostForMin)
+    end if
+  end align
+end Crabmada
+
+@main
+def main =
```

Comment:
In order to integrate with the website, you will need separate methods `def part1(input: String): Int` and `def part2(input: String): Int`.

Reply:
Thanks! Fixed!